File: backend/test_fav.py
import httpx
import asyncio

async def test_favorites():
    async with httpx.AsyncClient(base_url="http://localhost:8000") as client:
        # Add a favorite
        payload = {
            "content_type": "vod",
            "content_id": "test_123",
            "content_data": {"name": "Test Movie"}
        }
        res = await client.post("/api/user/favorites", json=payload)
        print("Add response:", res.json())
        
        # List favorites
        res = await client.get("/api/user/favorites")
        favs = res.json()
        print("Favorites list contains test_123?", any(f.get("content_id") == "test_123" for f in favs))
        
        # Remove favorite
        res = await client.delete("/api/user/favorites/vod/test_123")
        print("Delete response:", res.json())
        
        # List favorites again
        res = await client.get("/api/user/favorites")
        favs = res.json()
        print("Favorites list contains test_123 after delete?", any(f.get("content_id") == "test_123" for f in favs))

if __name__ == "__main__":
    asyncio.run(test_favorites())

File: frontend/test_fav.js
const axios = require('axios');

async function run() {
    const api = axios.create({ baseURL: 'http://localhost:8000/api' });

    // 1. Add
    console.log("Adding...");
    const addRes = await api.post("/user/favorites", {
        content_type: "vod",
        content_id: "659700",
        content_data: { name: "Test" }
    });
    console.log("Add:", addRes.data);

    // 2. Remove
    console.log("Removing...");
    const removeRes = await api.delete("/user/favorites/vod/659700");
    console.log("Remove:", removeRes.data);

    // 3. List
    const listRes = await api.get("/user/favorites");
    console.log("List has 659700?", listRes.data.some(f => f.content_id === "659700"));
}

run().catch(console.error);
